fecgov/fec-cms
# JS tests for campaign finance archive

## Summary
First pass as test coverage for campaign finance archive. 
 - Testing disableNonPresYears fucntion
 - Testing showTable function
 -  Istanbul html coverage reporter throws error for arrow functions that have a return as in statistical-summary-archive.js, so I am commenting out line 44 of karmaconf.js. This only effects the index.js coverage report that it generates and can be uncommented to generate that report.
 
This was an `innovation sprint` task to get spun up on JS testing. The next step is to add more tests for this and, more importantly, to add tests to the non-archive camp finance script to catch errors when a dev has to edit that script in future years. 


## Impacted areas of the application
modified:   fec/static/js/pages/statistical-summary-archive.js
new file: fec/static/js/modules/statistical-summary-archive.js
new file: fec/tests/js/statistical-summary-archive.js

## How to test

Include any information that may be helpful to the reviewer(s).
- checkout `tests-camp-fin-archive`
- `npm run build`
- `npm run test-single`
-  Can also test by changing the value on line 90 of the tests/ file from 1982 to 1978, this will fail the test.
- You can look at the page on prod to see the functionality when you choose `presidential`https://www.fec.gov/campaign-finance-data/campaign-finance-statistics-pre-1989/?year=1988&filer=congressional

____
